cylc/flow/config.py

```diff
@@ -2447,6 +2448,13 @@ def _get_taskdef(self, name: str) -> TaskDef:
 
         try:
             rtcfg = self.cfg['runtime'][name]
+
+            # If the workflow mode is simulation or dummy always
+            # override the task config:
+            workflow_run_mode = RunMode.get(self.options)
+            if workflow_run_mode.value in WORKFLOW_ONLY_MODES:
+                rtcfg['run mode'] = workflow_run_mode.value
```

Comment:
FYI broadcasting a run mode of live or skip seems to override this

Reply:
Correct right?

Reply:
Shouldn't be able to broadcast task run mode = live when the workflow is running in simulation mode? Not that it matters too much

Reply:
Ah right, yes. Unfortunately rather hard to prevent but will result in `run mode` being set (and reported) with an erroneous value :(

Reply:
I believe that I've now modified broadcast to ignore `[<namespace]run mode` and warn the user that run mode has been ignored if the workflow run mode is set to simulation or dummy.